langfuse/src/openai.ts

```diff
@@ -0,0 +1,277 @@
+            release: this.config?.release
+        })
+    }
+
+    createGeneration(
+        output?: any,
+        usage?: CompletionUsage,
+        error?: "DEBUG" | "DEFAULT" | "WARNING" | "ERROR" | undefined,
+        statusMessage?: string,
+        endTime = new Date()
+    ): void {
+        const input = parseInputArgs(this.inputs ?? {})
+        this.trace?.generation({
+            model: input.model,
+            name: this.config?.trace_name,
+            input: input.input,
+            modelParameters: input.modelParams,
+            startTime: this.startTime,
+            endTime: endTime,
+            level: error,
+            statusMessage,
+            output,
+            usage
+        })
+        this.trace?.update({ output })
+    }
+
+    async flush(): Promise<void> {
+        client.flush()
+    }
+}
+
+export const openaiTracer = async<T extends (...args: any[]) => any>(
+    baseFunction: T,
+    config?: WrapperConfig,
+    ...args: Parameters<T>
+): Promise<ReturnType<T> | any> => {
+    const tracer = new TraceGenerator({
+        name: baseFunction.name,
+        inputs: args[0],
+        config
+    })
+
+    tracer.createTrace()
+    let res: any;
+    try {
+        res = await baseFunction(...args)
+        if (isAsyncIterable(res)) {
+            async function* tracedOutputGenerator(): AsyncGenerator<unknown, void, unknown> {
+                const response = res
+                const chunks: unknown[] = [];
+                // TypeScript thinks this is unsafe
+                for await (const chunk of response as AsyncIterable<unknown>) {
+                    const _chunk = processChunks(chunk)
+                    chunks.push(_chunk);
+                    yield chunk;
+                }
+                tracer.createGeneration(chunks.join(""))
+                await tracer.flush()
+            }
+            return tracedOutputGenerator() as ReturnType<T>
+
+        } else {
+            tracer.createGeneration(
+                generateOutput(res),
+                getUsageDetails(res)
+            )
+            await tracer.flush()
+        }
+    } catch (error) {
+        tracer.createGeneration(
+            undefined,
+            undefined,
+            "ERROR",
+            String(error)
+        )
+        await tracer.flush()
+        throw error
+    }
+    return res
+}
+
+
+const isAsyncIterable = (x: unknown): x is AsyncIterable<unknown> =>
+    x != null &&
+    typeof x === "object" &&
+    typeof (x as any)[Symbol.asyncIterator] === "function";
+
+
+export const wrappedTracer = <T extends (...args: any[]) => any>(
+    baseFuntion: T,
+    config?: any
+): (...args: Parameters<T>) => Promise<ReturnType<T>> => {
+    {
+        try {
+            return (...args: Parameters<T>): Promise<ReturnType<T>> => openaiTracer(baseFuntion, config, ...args);
+        } catch (error) {
+            throw error
+        }
+    }
+}
+
+interface WrapperConfig {
```

Comment:
I think going for camelcase here would be a better fit to the rest of the JS SDK (even though the openai sdk uses snake case, what do you think?

Reply:
I agree. This makes sense. We use camel casing on all the other places and since this is our wrapper, it would make sense to keep it as camel cased.